internal/api/server_test.go

```diff
@@ -48,16 +56,277 @@ func (m *mockProbeStore) UpdateProbe(ctx context.Context, probe v1.ProbeObject)
 }
 
 func (m *mockProbeStore) ListProbes(ctx context.Context, selector string) ([]v1.ProbeObject, error) {
-	return nil, errors.New("not implemented")
+	if m.listProbesErr != nil {
+		return nil, m.listProbesErr
+	}
+	var res []v1.ProbeObject
+	for _, p := range m.probes {
+		res = append(res, p)
+	}
+	return res, nil
 }
```

Comment:
I think there are tools/packages we could potentially use to generate our mocks for us. If we use those, we could hook their invocation into the relevant make targets and offload the maintenance of these test structs to that

Reply:
On second thought, tools like https://github.com/uber-go/mock can be more trouble than they're worth, and it might add unnecessary complexity at this point.